rfcs/text/0000-tech-planning-process.md

```diff
@@ -0,0 +1,136 @@
+
+**Component:** Also known as a “Box” or a *chartered component*; represents a piece of the DCP led by a TL and worked on
+by a team of software engineers. The list of chartered components are maintained
+[here](https://github.com/HumanCellAtlas/dcp-community/tree/master/charters).
+
+
+## Detailed Design
+
+### Input to the TL Planning Process
+
+1) Date of TL+PM planning call (information to be gathered by the TL Roadmapping Shepherd).
+
+2) Epics/Objectives from previous quarter that will slip the release and rollover into the next.
+
+### Timeline
+
+Per the Roadmaps and Planning Process, a TL+PM call is expected to take place during the third week of Milestone 3 in a
+Release quarter.
+
+| &nbsp;&nbsp;Week in Milestone (Week in Quarter) &nbsp;&nbsp;| Deliverables |
+|           :----:                    | :--- |
+| **M2W3 (Week 7)**  | - Each component MUST enumerate the work they believe the DCP should prioritize in the following quarter, along with rough estimates of time (sized using T-shirt sizes) and specify any other activities that will also occur (e.g. onboarding new team members) and rough estimates of time. |
+| **M2W4 (Week 8)**  | - A 1 hour TL meeting (the TL Coalescing meeting) MUST occur to consolidate the objectives that have been individually presented and any projects involving multiple components will be sized. |
+|                    | - After the objectives have been coalesced, the union of all presented objectives (minus any objectives that were removed due to low priority) will be sent out to TLs to individually rank in priority. |
+| **M3W1 (Week 9)**  | - A second 1 hour TL meeting (the TL Prioritizing meeting) will take place to bring together all TLs rankings to narrow down the objectives list to objectives that are high priority and feasible. |
+| **M3W2 (Week 10)** | - The TL Planning Shepherd will turn the objectives agreed upon as representative of the TL priorities in the TL Prioritization meeting into an Information RFC and publish. This represents the final artifact: the **Technical Objectives RFC**. There will be no review period as the key stakeholders (the TLs) have already agreed upon the list. |
+| **M3W3 (Week 11)** | - A 2 hour meeting MUST occur with TLs and PMs “to review any incomplete Objectives in the Release, the next Objectives in the DCP Roadmap, and the ***Technical Objectives RFC***.” |
+
+**Note: MXWY is equal to “week Y in milestone X.”
+
+**Note: this schedule may differ based on the holiday schedule each quarter. The responsibility is on the Shepherd of
+the quarter to make sure the timeline is clear (especially the date of the TL+PM call) prior to the process commencing.
+
+### Templates
+
+For the purpose of curating the individual component objectives and for the consolidation of objectives into one that
+represents all the components’ objectives, please use: 
+[TL Objectives Template](https://docs.google.com/document/d/1jUo9Y7y9BJhVe8qpgljI45W7JpNlq3ownVBLA3wK0Qw/edit).
```

Comment:
In your template

1) "Top DCP Technical Objectives - Rollover from QX" is asking for a repeat of information that is already reported at the DCP level
2) As "Cross-box Technical Objectives" covers uncharted objectives I think it should be "Unchartered Technical Objectives" since "cross-box" is confusing with the cross-box work that is chartered.
3) Why does "Cross-box Technical Objectives" need to be broken out separately? Can we reduce clutter by calling out unchartered work in "Top DCP Technical Objectives - New for QX" where necessary?

Reply:
1. I was trying to get the template to encompass everything. I feel that it helps a lot to have everything in one place to really understand what's on table. Even if it is just copy-paste, at least it is easier to take a look at this single source and say "oof, there's a lot of rollover from the previous quarter, maybe we should cut down on these new features."

2. Done :)

3. I broke it out separately because sometimes folks have grand plans of work that is cross-functional but actually don't have resources allocated yet from other groups so separating them out makes it much more clear about requests for resources (and when I say resources, I largely mean time from engineers).

Reply:
Gonna say it now, for 1) I'll be just posting github links. DRY :)